lifelines/estimation.py

```diff
@@ -29,19 +31,173 @@ def __repr__(self):
         return s
 
 
-class ExponentialFitter(BaseFitter):
+class WeibullFitter(BaseFitter):
+
     """
-    This class implements an exponential model for univariate data. 
+    This class implements an Weibull model for univariate data. The model has parameterized
+    form:
+
+      S(t) = exp(-(lambda*t)**rho),   lambda >0, rho > 0,
+
+    which implies the cumulative hazard rate is
+
+      H(t) = (lambda*t)**rho,
+
+    and the hazard rate is:
 
-    S(t) = exp(-\lambda*t)
+      h(t) = rho*lambda(lambda*t)**rho
 
-    This implies a constant hazard rate of \lambda. 
+    After calling the `.fit` method, you have access to properties like:
+    `cumulative_hazard_', 'survival_function_', 'lambda_' and 'rho_'.
 
     """
 
+    def fit(self, durations, event_observed=None, timeline=None, entry=None,
+            label='Weibull_estimate', alpha=None, ci_labels=None):
+        """
+        Parameters:
+          duration: an array, or pd.Series, of length n -- duration subject was observed for
+          timeline: return the best estimate at the values in timelines (postively increasing)
+          event_observed: an array, or pd.Series, of length n -- True if the the death was observed, False if the event
+             was lost (right-censored). Defaults all True if event_observed==None
+          entry: an array, or pd.Series, of length n -- relative time when a subject entered the study. This is
+             useful for left-truncated observations, i.e the birth event was not observed.
+             If None, defaults to all 0 (all birth events observed.)
+          label: a string to name the column of the estimate.
+          alpha: the alpha value in the confidence intervals. Overrides the initializing
+             alpha for this call to fit only.
+          ci_labels: add custom column names to the generated confidence intervals
+                as a length-2 list: [<lower-bound name>, <upper-bound name>]. Default: <label>_lower_<alpha>
 
-    def __init__(self, alpha=0.95):
-        self.alpha = alpha
+        Returns:
+          self, with new properties like `cumulative_hazard_', 'survival_function_', 'lambda_' and 'rho_'.
+
+        """
+        self.durations = np.asarray(durations, dtype=float)
+        # check for negative or 0 durations - these are not allowed in a weibull model.
+        if np.any(self.durations <= 0):
+            filler = 0.01
+            print("Non-positive times found in durations. Replacing each non-positive value with %.5f" % filler)
```

Comment:
I think we should raise a ValueError or something in case of negative durations. A duration can't be negative. If the user is inputting negative durations (as I am), they are not inputting the correct thing.